api/tests/integration/ArticleTest.php

```diff
@@ -323,6 +324,57 @@ public function testPatchOneNewPermalink()
         $this->cleanupDiscussions([$entity]);
     }
 
+    public function testPatchOneExistingPermalinkSameEntity()
+    {
+        $factory = $this->getFactory(Article::class);
+        $entity = $factory->create();
+        $this->addPermalinksToArticles([$entity]);
+
+        $data = $factory->setTransformer(ArticleTransformer::class)
+            ->setModel($entity)
+            ->showOnly(['permalink'])
+            ->transformed();
+
+        $linksCount = $entity->articlePermalinks->count();
+
+        $this->patchJson('/articles/'.$entity->article_id, $data);
+        $this->shouldReturnJson();
+        $this->assertResponseStatus(204);
+
+        $checkEntity = Article::find($entity->article_id);
+        $this->assertEquals($checkEntity->permalink, $entity->permalink);
+        $this->assertEquals($checkEntity->articlePermalinks->count(), $linksCount);
+
+        $this->cleanupDiscussions([$entity]);
+    }
+
+    public function testPatchOneExistingPermalinkDifferentEntity()
+    {
+        $this->markTestSkipped('This test does not pass despite the assertion working in browser. @todo resolve reason');
```

Comment:
@Redjik can you please look in to this test and why it is not passing. I've had a good look through the logic, and it does seem like it should pass, however the check for the `unique` constraint does not seem to return any rows for some reason.

My suspicion is that there is something wrong with the permalink saving in the `Article` model event, but I can't see anything particularly wrong with the logic. As you implemented permalinks I'm hopeful you'll be able to shed light on the situation.